### PR TITLE
Features/vu 15 be share document a pi

### PR DESCRIPTION
**Pull request by huytran46**
- VU 15 BE - share document API

### CODE REVIEW

File: backend/src/models/account/account.model.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ const AccountSchema = new Schema<
 );
 
 AccountSchema.pre("save", function (next) {
-  var user = this;
```
Comment (vudoan1708-cyber) [inline backend/src/models/account/account.model.ts:32]: loz, nay thi dung var, hen la chu sua r chu neu k la bi chui sap mat

File: backend/src/models/document/document.model.ts [hunk 2]
```diff
@@ -34,12 +35,42 @@ const DocumentSchema = new Schema<
...
+      document.passcode = hash;
+
+      next();
+    });
+  });
+});
+
+DocumentSchema.method(
+  "comparePasscode",
+  async function comparePasscode(p: string) {
```
Comment (vudoan1708-cyber) [inline backend/src/models/document/document.model.ts:69]: dung fat arrow function di loz, long live ES6

**Review by vudoan1708-cyber — APPROVED** [review]

File: backend/src/models/account/account.model.ts (hunk 1 above)
Comment (huytran46) [reply backend/src/models/account/account.model.ts:32]: you bích

File: backend/src/models/document/document.model.ts (hunk 2 above)
Comment (huytran46) [reply backend/src/models/document/document.model.ts:69]: bích

**Pull request merged**